images/win/scripts/Installers/Install-MSYS2.ps1

```diff
@@ -0,0 +1,11 @@
+################################################################################
+##  File:  Install-MSYS2.ps1
+##  Desc:  Install MSYS2 via chocolaty
+################################################################################
+
+Import-Module -Name ImageHelpers
+
+$msys2Path = "C:\tools\msys64"
+choco install msys2 -y --params "/InstallDir:$msys2Path"
```

Comment:
I'm rather sure that the mingw toolchains can be installed through msys2 so ideally we do that here and remove the separate installation of mingw. Also, I wonder if it's possible to tell GitForWindows to just use the installed msys2/mingw rather than bring in it's own. Any thoughts?

Reply:
Also, how does this affect existing builds that depend on the currently installed mingw?

Reply:
another comment. if avoiding chocolatey is not substantially difficult, I think we should avoid it since waiting for choco packages adds another layer of delay and potential problems that we can't control. [This comment](https://github.com/actions/virtual-environments/issues/30#issuecomment-566275476) by @eine suggests it should be easy enough to just untar.

Reply:
Re using the tar, I created an [`Install-MSYS2.ps1`](https://github.com/MSP-Greg/msys2-action/blob/install-msys2/Install-MSYS2.ps1) script mentioned in [Issue #30 comment](https://github.com/actions/virtual-environments/issues/30#issuecomment-575688053). Also installs typical MSYS2 shell tools and basic 32 & 64 bit compilers.

Having used MSYS2 for a while, the main tar is not revised often, as the current release is dated 2019-May-24.  Hence, the more important code is the update code, which hopefully can be run any time images are updated...

Reply:
> Hence, the more important code is the update code, which hopefully can be run any time images are updated...

I beg to differ. Updating by default prevents users from testing their tools on a stable environment.

Reply:
@alepauly, MSYS2 installation is already implemented in the #355 PR. Could you please also take a look at this PR? If it looks good for you, we probably should merge #355 (after adding some tests), and remove MSYS2 installation from this PR. What do you think about it?